### DISCUSSION TITLE
Custom Objectives - MethodError

### DISCUSSION THREAD

**Discussion by villrv** [MilesCranmer/PySR#410] [Q&A]
Hi! I am hoping this is a simple mistake on my end...I am completely new to Julia!

I am unable to run the custom objective function [example](https://astroautomata.com/PySR/examples/#9-custom-objectives). When I run `model.fit`, I get the following error (truncated to what I think is the relevant bit):

```
RuntimeError: <PyCall.jlwrap (in a Julia function called from Python)
JULIA: MethodError: no method matching _method_instances(::Type{typeof(my_custom_objective)}, ::Type{Tuple{Node{Float32}, Dataset{Float32, Float32, Matrix{Float32}, Vector{Float32}, Nothing, NamedTuple{(), Tuple{}}, Nothing, Nothing, Nothing, Nothing}, Options{Int64, DynamicExpressions.OperatorEnumModule.OperatorEnum, false, Optim.Options{Float64, Nothing}, StatsBase.Weights{Float64, Float64, Vector{Float64}}}, Nothing}})
The applicable method may be too new: running in world age 44885, while current world is 54430.
```

My Julia version is 1.9.2, and I am on an M2 Mac. I've tried changing relevant versions of the python dependencies to no avail. 

Any advice appreciated!

**Comment by MilesCranmer** [comment]
Hi @villrv,
This looks like a bug in PySR. How are you running things? If you file an issue in this repo (https://github.com/MilesCranmer/PySR/issues) it will ask for the different info that will help me debug.
Thanks,
Miles

**Reply by villrv** [reply]
Thanks! Done here: https://github.com/MilesCranmer/PySR/issues/411
